FrxEditor.inc

```diff
@@ -421,18 +421,30 @@ class FrxEditor {
       'default');
   }
 
-
-  public function addParameters($parms_to_add) {
-    $parms_to_add = (array)$parms_to_add;
-    foreach ($parms_to_add as $parm) {
-      $parms= array();
-      $parms[$parm] = array('id' => $parm );
-      $this->setFrxHeader('parameters', 'parm',
-      $parms,
-      array('id', 'label', 'require', 'desc', 'data_source', 'data_field', 'type'),
-        'default');
-    }
-  }
+  /**
+  * Add report patameters
+  * @param array $parm
```

Comment:
documentation name `$param` does not match parameter name `$parms_to_add`.

Reply:
We also need to document what this array can look like.  In practice, the legacy function didn't take an array at all, but a string: `addParameters('newid')`.  We should still allow for this usage.  We want to be able to pass in one (or more?) parameters as `addParameters( array('id'=>'one', 'label'=>'One'), true )`.  Can I call `addParameters(array('one', 'two', 'three'))` to add three new parameters with ids of "one", "two" and "three"?  Can I call `addParameters( array( array('id'=>'one', 'label'=>'One'), array('id'=>'two', 'label'=>'Two') ), true )` to add two new parameters?